csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -92,12 +98,12 @@ void mapAllocationDomain(
   for (auto* ref_id : ref_alloc_domain) {
```

Comment:
It might be nice to comment here that this is a fast to slow loop, given the new `std::reverse` at line 81.